src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -335,13 +335,19 @@ private static bool IsOSVersionAtLeast(int major, int minor, int build, int revi
             {
                 return current.Minor > minor;
             }
-            if (current.Build != build)
+            // Unspecified build component is to be treated as zero
+            int currentBuild = current.Build == -1 ? 0 : current.Build;
+            build = build < 0 ? 0 : build;
```

Comment:
nit: both can use the same condition for consistency:
```suggestion
            int currentBuild = current.Build < 0 ? 0 : current.Build;
            build = build < 0 ? 0 : build;
```